File: src/model/paginator.rs
use serde::Deserialize;
use serde::Serialize;

/// This struct represents a paginator
#[derive(Serialize, Deserialize, Clone, Debug, Default)]
#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
pub struct Paginator {
    pub start_time: Option<u64>,
    pub end_time: Option<u64>,
    pub limit: Option<u64>,
    pub before: Option<String>,
    pub after: Option<String>,
}

File: bindings/csharp/Paginator.cs
namespace OpenLimits
{
    using System.Runtime.InteropServices;
    [StructLayout(LayoutKind.Sequential, Pack = 1)]
    public class Paginator
    {
        public readonly ulong startTime;
        public readonly ulong endTime;
        public readonly ulong limit;
        public readonly string before;
        public readonly string after;

        public Paginator(ulong startTime, ulong endTime, ulong limit, string before, string after)
        {
            this.startTime = startTime;
            this.endTime = endTime;
            this.limit = limit;
            this.before = before;
            this.after = after;
        }
    }
}